File: test_allodb.py
#!/usr/bin/env python
"""
Test script to verify allodb can be imported through rpy2
"""
import rpy2
from rpy2.robjects import r
from rpy2.robjects.packages import importr
import pandas as pd
from rpy2.robjects import pandas2ri

# Activate pandas conversion
pandas2ri.activate()

try:
    # Import the allodb package
    print("Importing allodb...")
    allodb = importr('allodb')
    print("allodb package imported successfully!")
    
    # Execute an R command to get some data from allodb
    print("Loading test data from allodb...")
    r('data("biomass_family", package="allodb")')
    
    # Convert R dataframe to pandas
    biomass_family = r('biomass_family')
    
    # Convert to pandas dataframe
    print("Converting to pandas dataframe...")
    biomass_family_pd = pandas2ri.rpy2py_dataframe(biomass_family)
    
    # Display head of dataframe
    print("\nSample data from allodb:")
    print(biomass_family_pd.head())
    
    print("\nTest completed successfully!")
    
except Exception as e:
    print(f"Error: {e}")
    exit(1) 

File: test_allodb.R
#!/usr/bin/env Rscript

# Script to test if allodb is correctly installed
message("Testing allodb installation...")

# Check if allodb is installed, attempt to install if missing
if (!require("allodb", quietly = TRUE)) {
  message("allodb package not found, attempting to install...")
  
  # Try to install via remotes if available
  if (require("remotes", quietly = TRUE)) {
    remotes::install_github("ropensci/allodb")
  } else {
    # Try to install remotes first
    install.packages("remotes", repos = "https://cloud.r-project.org/")
    if (require("remotes", quietly = TRUE)) {
      remotes::install_github("ropensci/allodb")
    } else {
      message("Error: Failed to install remotes package")
      quit(status = 1)
    }
  }
  
  # Check if installation succeeded
  if (!require("allodb", quietly = TRUE)) {
    message("Error: Failed to install allodb package")
    quit(status = 1)
  }
} else {
  message("Success: allodb package loaded successfully")
}

# Test functionality
message("Available datasets in allodb:")
data(package = "allodb")

message("Loading sample data from allodb...")
data("missing_values", package = "allodb")
message("Sample data from allodb:")
print(head(missing_values))

message("Test completed successfully") 